src/Microsoft.Data.Sqlite.Core/SqliteConnection.cs

```diff
@@ -265,6 +265,32 @@ protected override DbCommand CreateDbCommand()
             => CreateCommand();
 
         /// <summary>
+        /// Create custom collation.
+        /// </summary>
+        /// <param name="name">Name of the collation.</param>
+        /// <param name="comparison">Method that compares two strings.</param>
+        public virtual void CreateCollation(string name, Comparison<string> comparison)
```

Comment:
Passing a `null` comparison will currently result in a `NullReferenceException`. We should either:
1. Throw `ArgumentNullException`
2. Support it by flowing `null` into `sqlite3_create_collation` (which removes the collation)